test/vm_tests/builder/parsing/ast_util_test.dart

```diff
@@ -152,17 +152,28 @@ main() {
           foo.SomeTypeName foo;
         ''')).nameWithoutPrefix, 'SomeTypeName');
       });
+
+      test('nameWithPrefix', () {
+        expect(TypeNameHelper(parseAndGetSingleWithType('''
+          SomeTypeName foo;
+        ''')).nameWithPrefix, 'SomeTypeName');
+
+        expect(TypeNameHelper(parseAndGetSingleWithType('''
+          foo.SomeTypeName foo;
+        ''')).nameWithPrefix, 'foo.SomeTypeName');
+      });
     });
 
     group('NameHelper', () {
```

Comment:
This extension is exclusive to `Identifier`, and in analyzer 6 type names are no longer parsed as `Identifier`, so these cases needed to be updated